lib/services/user_service.ex

```diff
@@ -12,6 +12,16 @@ defmodule SlackCoder.Services.UserService do
     if db_user, do: {:ok, user}, else: {:new, user}
   end
 
+  def save(user, changeset) do
+    if changeset.changes[:slack] || changeset.changes[:github] do
```

Comment:
You can also handle this with pattern matching...

You want to call `stop_user` whenever one of those two are present... so change it to something like:

```elixir
def save(user, changeset) do
  pre_save(changeset.changes, changeset.data)
  changeset
  |> Repo.update()
  |> SlackCoder.Users.Supervisor.start_user()
end

defp pre_save(%{slack: _}, user), do: SlackCoder.Users.Supervisor.stop_user(user)
defp pre_save(%{github: _}, user), do: SlackCoder.Users.Supervisor.stop_user(user)
defp pre_save(_, _user), do: nil
```